### PR TITLE
Update the multigate example (add performance and remove dwavebinarycsp)

### PR DESCRIPTION
**Pull request by JoelPasvolsky**

### CODE REVIEW

File: docs/examples/multi_gate.rst [hunk 1]
```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
...
-
-.. note:: `dwavebinarycsp` works best for constraints of up to 4 variables; it may not
-      function as expected for constraints of over 8 variables.
-
-The next line of code converts the constraints into a BQM that you solve by sampling.
-
-.. testcode::
+>>> from dimod.generators import and_gate, or_gate
+...
+>>> bqm2 = or_gate('b', 'c', 'out2')
```
Comment (arcondello) [inline docs/examples/multi_gate.rst:111]: why not name these functionally? Like `or_bc = or_gate('b', 'c', 'out2')` or whatever

File: docs/examples/multi_gate.rst [hunk 2]
```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
...
 
-.. toctree::
-   :maxdepth: 1
-   :hidden:
+>>> from dimod import ExactSolver
+...
+>>> solutions = ExactSolver().sample(bqm).lowest()
+>>> z = []
+>>> out_fields = [key for key in list(next(solutions.data(['sample'])))[0].keys() if 'out' in key]
+>>> for datum in solutions.data(['sample', 'energy']):
```
Comment (arcondello) [inline docs/examples/multi_gate.rst:132]: Why not just `print(solutions)` and use the pretty printing?

File: docs/examples/multi_gate.rst [hunk 3]
```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
...
+  ...    if sampleset.first.energy > 0:
+  ...        samplesets[runs].append(0)
+  ...    else:
+  ...        samplesets[runs].append(sum(sampleset.lowest().record.num_occurrences))
+  ...        embedding = sampleset.info["embedding_context"]["embedding"]  # doctest: +SKIP
+  ...        chain_len.append(max(len(x) for x in embedding.values()))     # doctest: +SKIP
+  ...        sampler2 = FixedEmbeddingComposite(qpu)), embedding=embedding)  # doctest: +SKIP
+  ...    for i in range(submission_repeats):
+  ...        print("\tSubmitting {}--{}".format(runs, i))
+  ...        sampleset = sampler2.sample(bqm, num_reads=5000, return_embedding=True)   # doctest: +SKIP
```
Comment (hhtong) [inline docs/examples/multi_gate.rst:594]: `sampler2` isn't defined here if `sampleset.first.energy > 0`

File: docs/examples/multi_gate.rst [hunk 4]
```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
...
+   clique-embedding (bottom-right).
+
+Short chains of a few qubits generally enable good quality solutions: as shown
+in the table above, for a two-replications circuit the ratio of ground states is
+similar for both embedding methods.
+
+The BQM representing ten replications of the
+:math:`z = \overline{b} (ac + ad + \overline{c}\overline{d})` circuit, for which
+a clique embedding of over 100 nodes is required, the clique embedding has much
+longer chains than direct embedding of the problem (11 versus 2 qubits).
```
[the remark this answers is not in the document]
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:478]: Rewrote, thanks

File: docs/examples/multi_gate.rst (hunk 2 above)
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:132]: The internal variables double the variable count, so it's harder to see ``z=func(a, b, c, d)`` in the output

File: docs/examples/multi_gate.rst (hunk 1 above)
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:111]: It makes an easy connection between the BQM for a gate and that gate's numbering in the figure above it (e.g., the users sees "bqm4" and then looks up the gate with a big "4" on it in the figure) 
The functionality is easily readable from the rhs (operation in the gate name and inputs in the arguments) so adding those to the lhs wouldn't add helpful info. 
If this content moves in the future somewhere that separates code from description (e.g., an example with a readme or JN) then functional names would be better, but close on the same page, this makes a nice connection

Comment (arcondello) [reply docs/examples/multi_gate.rst:111]: Sure, defer to what you think makes sense. Though IMO if that's the reasoning I might name them `gate4 = or_gate(...)`

File: docs/examples/multi_gate.rst (hunk 2 above)
Comment (arcondello) [reply docs/examples/multi_gate.rst:132]: A good future use case for https://github.com/dwavesystems/dimod/pull/806 though IMO

Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:132]: Oh, that's a nice PR

File: docs/examples/multi_gate.rst (hunk 1 above)
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:111]: Sure, I updated to ``bqm_gate<num gate>`` (I want to also keep the "bqm" to emphasize that they are creating BQMs and then adding them to a comprehensive BQM

File: docs/examples/multi_gate.rst (hunk 2 above)
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst]: Reimplemented using https://github.com/dwavesystems/dimod/pull/806 -- anything else needed for approval & merge?

**Review by randomir — APPROVED** [review]
Lovely.

**Review by arcondello — APPROVED** [review]

**Comment by randomir** [comment]
@JoelPasvolsky, for doctest to succeed, you'll need to require `dimod>=0.10.9` in test or docs requirements.

File: docs/examples/multi_gate.rst [hunk 5]
```diff
@@ -88,269 +90,517 @@ To run the code in this example, the following is required.
...
+it to visualize sample sets returned from a quantum computer in your browser.
 
-You can see the minor-embeddings used here: :ref:`multi_gate_results`; below
-those embeddings are visualized graphically.
-
-.. figure:: ../_images/MultiCSPembeddings.png
-   :name: MultiCSPembeddings
+>>> import dwave.inspector
+>>> bqm = circuit_bqm(2)
+>>> sampleset1 = sampler1.sample(bqm, num_reads=5000, return_embedding=True)
```
[the remark this answers is not in the document]
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:457]: Removed for this instance

**Comment by JoelPasvolsky** [comment]
CI is failing for https://github.com/dwavesystems/dwave-ocean-sdk/pull/168 on a dwave-hybrid unittest

**Pull request merged**